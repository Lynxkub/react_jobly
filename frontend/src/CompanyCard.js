import React from 'react';



const CompanyCard = ({company}) => {
    
    return(
        <div className='card bg-dark' style={{'width' : '100vw'}}>
            <div className='card-body'>
            <h3 className='card-title text-light'>{company.name}</h3>
            <p className='card-text text-light'>{company.description}</p>
            <p className='card-text text-light'>Employees : {company.numEmployees}</p>
            {/* {company.logoUrl !== null ? <div><img src = {company.logoUrl} /></div> : null} */}
            </div>
        </div>
    )
}

export default CompanyCard;