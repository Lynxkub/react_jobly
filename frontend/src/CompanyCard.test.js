import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import {render , fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import CompanyCard from './CompanyCard';

it('renders without crashing' , () => {
    
    const company = {
        name : 'test',
        description: 'test',
        numEmployees : 10,
        logoUrl : null
    }

    render(<CompanyCard company = {company}/>)
})

it('matches snapshot' , () => {
    const company = {
        name : 'test',
        description: 'test',
        numEmployees : 10,
        logoUrl : null
    }
    const {asFragment} = render(<CompanyCard company = {company}/>);
    expect(asFragment()).toMatchSnapshot();
})

it('allows a click to bring to a company page that displays jobs' , () => {
    const company = {
        name : 'test',
        description: 'test',
        numEmployees : 10,
        logoUrl : null,
        jobs : [
            {
                id: 1 , 
                title: 'test',
                salary : 100,
                equity : '0.99'
            }

        ]
    }

    const {getByText} =render(<CompanyCard company = {company} />);
    setTimeout(() => {
        const name = getByText('test');
        expect(name).toBeInTheDocument();
        fireEvent.click(name);
            setTimeout(() => {
            // check to see if the link to the company page is loaded and a job card is shown with the correct equity piece
            const jobEquity = getByText('0.99')
            expect(jobEquity).toBeInTheDocument();
        } , 1000)
    } , 1000)
    

})

