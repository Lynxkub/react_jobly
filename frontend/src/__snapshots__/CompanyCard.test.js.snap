// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      test
    </h3>
    <p>
      test
    </p>
    <p>
      10
    </p>
  </div>
</DocumentFragment>
`;
